src/dappsExplorer/DAppsExplorerScreenSearch.tsx

```diff
@@ -0,0 +1,326 @@
+import { useDispatch, useSelector } from 'react-redux'
+import { DappExplorerEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import SearchInput from 'src/components/SearchInput'
+import {
+  dappFavoritesEnabledSelector,
+  dappsCategoriesAlphabeticalSelector,
+  dappsListErrorSelector,
+  dappsListLoadingSelector,
+  dappsListSelector,
+  dappsMinimalDisclaimerEnabledSelector,
+  favoriteDappIdsSelector,
+} from 'src/dapps/selectors'
+import { fetchDappsList } from 'src/dapps/slice'
+import { DappSection, DappV1, DappV2 } from 'src/dapps/types'
+import DappCard from 'src/dappsExplorer/DappCard'
+import FavoriteDappsSectionSearch from 'src/dappsExplorer/FavoriteDappsSectionSearch'
+import NoResultsSearch from 'src/dappsExplorer/NoResultsSearch'
+import useDappFavoritedToast from 'src/dappsExplorer/useDappFavoritedToast'
+import useDappInfoBottomSheet from 'src/dappsExplorer/useDappInfoBottomSheet'
+import useOpenDapp from 'src/dappsExplorer/useOpenDapp'
+import { calculateSearchScore } from 'src/dappsExplorer/utils'
+import { currentLanguageSelector } from 'src/i18n/selectors'
+import Help from 'src/icons/Help'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { useDebounce } from 'use-debounce'
+
+const AnimatedSectionList =
+  Animated.createAnimatedComponent<SectionListProps<DappV2, SectionData>>(SectionList)
+
+interface SectionData {
+  data: DappV2[] | DappV1[]
+  category: string
+}
+
+export function DAppsExplorerScreenSearch() {
+  const { t } = useTranslation()
+  const insets = useSafeAreaInsets()
+
+  const sectionListRef = useRef<SectionList>(null)
+  const scrollPosition = useRef(new Animated.Value(0)).current
+
+  const onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: scrollPosition } } }])
+  const dispatch = useDispatch()
+  const loading = useSelector(dappsListLoadingSelector)
+  const error = useSelector(dappsListErrorSelector)
+  const categories = useSelector(dappsCategoriesAlphabeticalSelector)
+  const dappFavoritesEnabled = useSelector(dappFavoritesEnabledSelector)
+  const dappsMinimalDisclaimerEnabled = useSelector(dappsMinimalDisclaimerEnabledSelector)
+  const dappList = useSelector(dappsListSelector)
+  const language = useSelector(currentLanguageSelector)
+  const favoriteDappsById = useSelector(favoriteDappIdsSelector)
+
+  // Some state lifted up from all and favorite sections
+  const [searchQuery, setSearchQuery] = React.useState('')
+  // @ts-expect-error 'previousSearchQuery' is declared but its value is never read
+  const [previousSearchQuery, setPrevSearchQuery] = React.useState('')
+  const [value] = useDebounce(searchQuery, 1000)
+  const [previousValue] = useDebounce(value, 1000)
```

Comment:
I think I need at least 3 of the 4 too keep the analytics correct and the fourth to avoid build errors 🤔 

#### Example: User enters 'Eth'
----- E -----
`value=''`
`previousValue=''`
`searchQuery='E'`
`previousSearchQuery=''`

----- t -----
`searchQuery='Et'`
`previousSearchQuery='E'`

----- h -----
`searchQuery='Eth'`
`previousSearchQuery='Et'`

----- 1000 MS -----
`value='Eth'`
Analytics for search submitted:
`ValoraAnalytics.track(DappExplorerEvents.dapp_search, { query: 'Eth', clearing: false })`

----- 1000 MS -----
`previousValue='Eth'`

----- User Clears Input -----
`previousSearchQuery='Eth'`
`searchQuery=''`

----- 1000 MS -----
`value=''`
Analytics for search submitted:
`ValoraAnalytics.track(DappExplorerEvents.dapp_search, { query: 'Eth', clearing: true })`